packages/core/utils/src/dml/__tests__/entity-builder.spec.ts

```diff
@@ -754,6 +850,112 @@ describe("Entity builder", () => {
         },
       })
     })
+
+    test("define index when entity is using an explicit pg Schema", () => {
+      const model = new EntityBuilder()
+      const user = model.define("platform.user", {
+        id: model.number().index(),
+        username: model.text(),
+        email: model.text().unique(),
+      })
+
+      const entityBuilder = createMikrORMEntity()
+      const User = entityBuilder(user)
+      expectTypeOf(new User()).toMatchTypeOf<{
+        id: number
+        username: string
+        email: string
+        deleted_at: Date | null
+      }>()
+
+      const metaData = MetadataStorage.getMetadataFromDecorator(User)
+      expect(metaData.className).toEqual("User")
+      expect(metaData.path).toEqual("User")
+
+      expect(metaData.indexes).toEqual([
+        {
+          name: "IDX_users_id",
+          expression:
+            'CREATE INDEX IF NOT EXISTS "IDX_users_id" ON "platform"."users" (id) WHERE deleted_at IS NULL',
```

Comment:
todo: the table name should be singular, this is the convention we went with

Reply:
Ohh okay! Usually I have seen it to be plural

Reply:
yeah me too, but medusa convention is singular :D